File: examples/python/django/recoverability.py
def process_delivery(delivery_id: int) -> None:
    delivery = WebhookDelivery.objects.get(id=delivery_id)

    try:
        webhook_pipeline.process_delivery(delivery)
    except TemporaryWebhookError as exc:
        delivery.mark_retryable(str(exc))
        retry_webhook_delivery.apply_async(args=[delivery.id], countdown=60)
    except Exception as exc:
        delivery.mark_failed(str(exc))
        raise
    else:
        delivery.mark_completed()

File: examples/go/net-http/recoverability.go
package examples

import (
	"context"
	"errors"
	"time"
)

func (w *Worker) ProcessDelivery(ctx context.Context, deliveryID string) error {
	delivery, err := w.DeliveryStore.Get(ctx, deliveryID)
	if err != nil {
		return err
	}

	if err := w.Pipeline.ProcessDelivery(ctx, delivery); err != nil {
		if errors.Is(err, ErrTemporaryWebhookFailure) {
			_ = w.DeliveryStore.MarkRetryable(ctx, deliveryID, err.Error())
			return w.RetryQueue.Schedule(ctx, deliveryID, time.Minute)
		}

		_ = w.DeliveryStore.MarkFailed(ctx, deliveryID, err.Error())
		return err
	}

	return w.DeliveryStore.MarkCompleted(ctx, deliveryID)
}
